apps/web/src/components/Basenames/UsernameProfileContext/index.tsx

```diff
@@ -1,61 +1,126 @@
 'use client';
 import { BaseName } from '@coinbase/onchainkit/identity';
+import { USERNAME_L2_RESOLVER_ADDRESSES } from 'apps/web/src/addresses/usernames';
+import useBaseEnsName from 'apps/web/src/hooks/useBaseEnsName';
+import useBasenameChain from 'apps/web/src/hooks/useBasenameChain';
+import { buildBasenameOwnerContract } from 'apps/web/src/utils/usernames';
 import {
   Dispatch,
   ReactNode,
   SetStateAction,
   createContext,
+  useCallback,
   useContext,
   useMemo,
   useState,
 } from 'react';
 import { Address } from 'viem';
-import { useAccount } from 'wagmi';
+import { useAccount, useEnsAddress, useReadContract } from 'wagmi';
 
 export enum UsernameProfileSteps {}
 
 export type UsernameProfileContextProps = {
+  // Profile details
   profileUsername: BaseName;
   profileAddress?: Address;
-  currentWalletIsOwner?: boolean;
+
+  // Profile owner
+  profileOwnerAddress?: Address;
+  profileOwnerUsername?: BaseName;
+  profileOwnerRefetch: () => Promise<void>;
+
+  // State
+  currentWalletIsProfileOwner: boolean;
+  currentWalletIsProfileAddress: boolean;
+
+  // Settings
   showProfileSettings: boolean;
   setShowProfileSettings: Dispatch<SetStateAction<boolean>>;
 };
 
 export const UsernameProfileContext = createContext<UsernameProfileContextProps>({
   profileUsername: 'default.basetest.eth',
   profileAddress: undefined,
-  currentWalletIsOwner: false,
+  profileOwnerAddress: undefined,
+  profileOwnerRefetch: async () => undefined,
```

Comment:
this feels like an unnatural default value

Reply:
undefined feels more expected

Reply:
so we either defined it weirldy here, or we do `profileOwnerRefetch.()` everywhere it's being used. I prefer the first but not feeling strongly. 

typed context are a pain for this, without default values matching the `UsernameProfileContextProps` it will cry.

Reply:
I don't think it's that weird, Lodash has noop for similar usecases https://lodash.com/docs/#noop

Reply:
Maybe add an explicit noop util and import it wherever this pattern is needed. That would make me happier about it.

```
function noop() { }
```